paasta_tools/autoscaling/autoscaling_cluster_lib.py

```diff
@@ -162,19 +190,11 @@ def spotfleet_metrics_provider(spotfleet_request_id, resource, pool_settings):
         for pair in zip(free_pool_resources, total_pool_resources)
     ])
     target_utilization = pool_settings.get('target_utilization', DEFAULT_TARGET_UTILIZATION)
-    error = utilization - target_utilization
-    current, target = get_spot_fleet_delta(resource, error)
-    return current, target
+    return utilization - target_utilization
 
 
 def get_spot_fleet_delta(resource, error):
-    ec2_client = boto3.client('ec2', region_name=resource['region'])
-    spot_fleet_request = ec2_client.describe_spot_fleet_requests(
-        SpotFleetRequestIds=[resource['id']])['SpotFleetRequestConfigs'][0]
-    if spot_fleet_request['SpotFleetRequestState'] != 'active':
-        raise ClusterAutoscalingError('Can not scale non-active spot fleet requests. This one is "%s"' %
-                                      spot_fleet_request['SpotFleetRequestState'])
-    current_capacity = int(spot_fleet_request['SpotFleetRequestConfig']['TargetCapacity'])
+    current_capacity = int(resource['sfr']['SpotFleetRequestConfig']['FulfilledCapacity'])
```

Comment:
This is actually quite a "big" change but I think a correct one.
Currently:
* We get the current target from AWS.
* Calculate the percentage error based on mesos stats on cpu/ram etc.
* Set a new target based on that error
* Kill instances and count down to the new target from the old target, stopping when we get close.

Now:
* Get current fulfilled by AWS
* Calculate the percentage error based on mesos stats on cpu/ram etc.
* Set a new target based on that error
* Kill instances and count down to the new target from the current fulfilled capacity, stopping when we get close.